android-mobile/src/org/subsurfacedivelog/mobile/AndroidSerial.java

```diff
@@ -0,0 +1,288 @@
+package org.subsurfacedivelog.mobile;
+
+import com.hoho.android.usbserial.driver.UsbSerialDriver;
+import com.hoho.android.usbserial.driver.UsbSerialPort;
+import com.hoho.android.usbserial.driver.UsbSerialProber;
+
+import android.hardware.usb.UsbDeviceConnection;
+import android.hardware.usb.UsbManager;
+import android.hardware.usb.UsbDevice;
+
+import android.content.Context;
+import android.util.Log;
+import android.app.PendingIntent;
+import android.content.Intent;
+
+import org.subsurfacedivelog.mobile.SubsurfaceMobileActivity;
+
+import java.lang.System;
+import java.lang.Thread;
+import java.util.Queue;
+import java.util.List;
+import java.util.LinkedList;
+import java.lang.Math;
+
+public class AndroidSerial {
+
+	private static final String TAG = "AndroidSerial";
+
+	private static final int DC_STATUS_SUCCESS = 0;
+	private static final int DC_STATUS_DONE = 1;
+	private static final int DC_STATUS_UNSUPPORTED = -1;
+	private static final int DC_STATUS_INVALIDARGS = -2;
+	private static final int DC_STATUS_NOMEMORY = -3;
+	private static final int DC_STATUS_NODEVICE = -4;
+	private static final int DC_STATUS_NOACCESS = -5;
+	private static final int DC_STATUS_IO = -6;
+	private static final int DC_STATUS_TIMEOUT = -7;
+	private static final int DC_STATUS_PROTOCOL = -8;
+	private static final int DC_STATUS_DATAFORMAT = -9;
+	private static final int DC_STATUS_CANCELLED = -1;
```

Comment:
that is of course rather fragile... would it be possible to auto-generate the code snippet here from libdivecomputer sources? Certainly not required to merge this, but might be a way to ensure that things don't break if Jef changes a constant.

Reply:
Huh, maybe that's possible, but I wouldn't know how...